sagemaker/pipeline/execute_sm_pipeline.py

```diff
@@ -0,0 +1,275 @@
+"""
+Copyright Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Execute a SageMaker pipeline for GraphStorm.
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+import warnings
+
+import boto3
+import psutil
+from sagemaker.workflow.pipeline import Pipeline
+from sagemaker.workflow.pipeline_context import LocalPipelineSession, PipelineSession
+
+from create_sm_pipeline import GraphStormPipelineGenerator
+from pipeline_parameters import load_pipeline_args
+
+
+def parse_args():
+    """Parse pipeline execution arguments"""
+    parser = argparse.ArgumentParser(
+        description="Execute GraphStorm SageMaker Pipeline"
+    )
+
+    parser.add_argument(
+        "--pipeline-name",
+        type=str,
+        required=True,
+        help="Name of the pipeline to execute. Required.",
+    )
+    parser.add_argument(
+        "--region",
+        type=str,
+        required=False,
+        help="AWS region. Required for SageMaker execution.",
+    )
+    parser.add_argument(
+        "--async-execution",
+        action="store_true",
+        help="Run pipeline asynchronously on SageMaker, return after printing execution ARN.",
+    )
+    parser.add_argument(
+        "--local-execution",
+        action="store_true",
+        help="Use a local pipeline session to execute the pipeline.",
+    )
+    parser.add_argument(
+        "--pipeline-args-json-file",
+        type=str,
+        help=(
+            "When executing locally, optionally provide a JSON representation of the pipeline arguments. "
+            "By default we look for '<pipeline-name>-pipeline-args.json' in the working dir."
+        ),
+    )
+
+    overrides = parser.add_argument_group(
+        "Pipeline overrides", "Override default pipeline parameters at execution time."
+    )
+
+    # Optional override parameters
+    overrides.add_argument("--instance-count", type=int, help="Override instance count")
```

Comment:
How to control the instance cnt for train/infer and gsprocessing separately?

Reply:
We have the pipeline parameter `gsprocessing_instance_count` but we only expose it during pipeline deployment. I can make that into an execution-time parameter as well.